src/main/java/use_cases/add_edit_category_use_case/CategoryUCI.java

```
@@ -3,52 +3,47 @@
 import entities.Category;
 import entities.EntityException;
 import entities.MonthlyStorage;
-import views.add_edit_category_views.CategoryP;
 
 import java.util.ArrayList;
 import java.util.NoSuchElementException;
 import java.util.Objects;
 
 /**
- * The Category_Use_Case_Interactor 1. adds/creates a new category and 2. edits an existing category. (Updates MonthlyStorage?)
+ * The Category_Use_Case_Interactor 1. adds/creates a new category and 2. edits an existing category.
+ * Updates MonthlyStorage:
+ * 1. adding a new Category to MonthlyStorage.categoryData with MonthlyStorage.addCategory()
+ * 2.
  * Contains a helper method isValidDouble() and findCategory().
  */
 
 public class CategoryUCI implements CategoryIB {
     final CategoryOB categoryOB;
-    private CategoryOD categoryOD;
 
     /**
      * Constructs CategoryUCI.
      * @param categoryP presenter that is related to the use case.
      */
 
-    public CategoryUCI(CategoryP categoryP) {
+    public CategoryUCI(CategoryOB categoryP) {
         this.categoryOB = categoryP;
-        this.categoryOD = null;
     }
 
     /**
-     * Helper method returns a boolean to check if the input Object is a valid double.
+     * Helper method returns a double to check if the input Object is a valid double.
      * @param value a user input
```

I believe the new toDouble just converts to double instead of checking if its a double. Also, does it not throw exception if the input isn't valid nuber?

---

According to this website:[https://www.geeksforgeeks.org/double-parsedouble-method-in-java-with-examples/](url), Double.parseDouble() may throw two exceptions which are NumberFormatException and NullPointerException. (The latter I just found out so I'll add it in.) 
I'm catching them in the addCategoryMonth() since I didn't have it in my helper. (This is a choice bc I thought so this way my addCategoryMonth() handles all the "failed" situations with their fail messages.) And before the InputMismatchException can happen due to an invalid type error, the input should either return a double or be caught with the exceptions mentioned above. Let me know if that makes sense.

---

I don't see how it would throw a InputMismatchException? Since every object can be turned into a string, the only thing that could throw is the Double.parseDouble method, but that one should throw a NumberFormatException.

---

Yes, the Double.parseDouble does throw the NumberFormatException and it is handled. What Yin is mentioning is the InputMismatchException thrown when in CategoryID when setValue() passes in a non-double type. But that shouldn't happen because before the object that fails to convert to double reaches the setter, a NumberFormatException is handled!

---

Also, I want to discuss whether to let it throw a NullPointerException as well. As how the link mentioned, parseDouble() can throw a NullPointerException when it passes in a null. However, I think NullPointerException is a RunTimeException and shouldn't be handled. In our case, the parseDouble() passes in user input and I don't think it's going to run if the input data is incomplete? (Correct if I'm wrong lol) Hence, I think it will be okay if we don't catch NullPointerException?

---

Nevermind, yeah, you are right. She assigns the result to a double variable, but the NumberFormatException will be caught before she does that.